### ISSUE TITLE
5.9.0-beta - Type 'Buffer<ArrayBuffer>' is not assignable to type 'ArrayBuffer'

### ISSUE THREAD

**Issue by maschwenk** [microsoft/TypeScript#62109]
### 🔎 Search Terms

ArrayBuffer
Type 'Buffer<ArrayBuffer>' is not assignable to type 'ArrayBuffer'.`
5.9.0-beta

### 🕗 Version & Regression Information

https://stackblitz.com/edit/vitejs-vite-ymeftbvq?file=tsconfig.json

With the exact same `@types/node` as `5.8.3`

```ts
const test: ArrayBuffer = Buffer.from('test', 'utf-8');
```
errors with `error TS2322: Type 'Buffer<ArrayBuffer>' is not assignable to type 'ArrayBuffer'.`

- This changed between versions 5.8.3 and 5.9.0-beta

### ⏯ Playground Link

https://stackblitz.com/edit/vitejs-vite-ymeftbvq?file=tsconfig.json

### 💻 Code

```ts
const test: ArrayBuffer = Buffer.from('test', 'utf-8');
```

### 🙁 Actual behavior

Failed typechecking

### 🙂 Expected behavior

Expected this to pass

**Comment by jakebailey** [comment]
Given Buffer isn't an ArrayBuffer, I'm pretty sure this is intentional. https://nodejs.org/api/buffer.html#bufbuffer is an ArrayBuffer.

Unless I'm really missing something.

**Comment by maschwenk** [comment]
Based on a private Discord discussion, this appears to be intentional, though I am curious if this should be flagging with `@types/node` 20.19.x. Also just curious which commit/PR changed this, I tried scanning through https://github.com/microsoft/TypeScript/compare/v5.8.3...v5.9-beta but was having trouble finding.

**Comment by jakebailey** [comment]
#61647 and related PRs.

All currently supported Node types pass with the last 10 versions of TS, so I would think this would behave correctly.

**Comment by Renegade334** [comment]
The quirk of the array buffer view (TypedArray/DataView) interfaces being assignable to `ArrayBuffer` was a complete accident previously. Mistaking array buffer views for `ArrayBuffer`s is a very common mishap, and one that the checker was previously blind to.

The divergence between the `ArrayBuffer` and array buffer view interfaces [has existed since TS5.7](https://www.typescriptlang.org/play/?target=99&ts=5.7.3#code/PTAEAEBcEMCcHMCmkBcpEGcB2iAekAoAgYwHssNJQAjAVwDN7FY0BBWWaATwCEGnYoALygcAd1ABVAJZZIADnacuACgCUAbiA), but only previously applied if targeting es2024+. Since #60150, it now applies universally.

**Comment by petamoriken** [comment]
Node.js's `Buffer` does NOT inherit from `ArrayBuffer`, but rather inherits from `Uint8Array`, so I think the problem is due to incorrect usage in the first place.

```ts
const test: Uint8Array<ArrayBuffer> = Buffer.from('test', 'utf-8');
```

**Comment by maschwenk** [comment]
Thanks all! Appreciate all that context!
